lib/mongo-stream.js

```diff
@@ -25,8 +27,18 @@ MongoStream.prototype._write = function (chunk, encoding, done) {
 
     dao.writeLog(data, function (err) {
       if (err) {
-        console.error('fh-bunyan failed to write log to mongodb', err);
-        console.error((err && err.stack) ? err.stack : 'no stack available');
+        if (isMongoConnectionError(err) && !flag) {                    //if the error is caused by the unavailability of mongodb
+          console.error('fh-bunyan failed to write log to mongodb', err);
```

Comment:
You're still logging on every code branch. This means that you're logging on every error. You need to have a branch of the code where you're not logging.